manifests/plugin/pulp.pp

```diff
@@ -19,6 +19,8 @@
 #
 # $client_authentication:: An array of client authentication types supported by the Pulp installation.
 #
+# $rhsm_url::              The URL to the Katello API for RHSM traffic or reverse proxy to it
```

Comment:
Can't we simplify this and state the intent?
```suggestion
# $rhsm_url::              The RHSM base URL. Katello will direct clients like subscription-manager here.
```

Reply:
I wrote a new version that tries to state the basic facts using accurate terminology, and follow up with an explanation of the intent:

`The RHSM base URL represents the URL to the Katello API for RHSM traffic, or a reverse proxy to it. Katello will direct Content Hosts registering through this SmartProxy server to send subscription-manager traffic to this URL.`

I didn't include the language of `clients like subscription-manager` because to my knowledge subscription-manager is the only RHSM client that Katello supports.

Let me know your thoughts please.

Reply:
My point was to avoid "Katello API for RHSM traffic, or a reverse proxy to it". A URL is IMHO sufficient abstraction. The important bit of information is that we are directing clients to this URL. How that URL is served is irrelevant in this module. If at some point we implement it in a way that we directly send clients to Candlepin, we will still direct clients to the URL specified here.

Put in another way: we should describe what this value represents and how/what it's used for. Not how it should be implemented.

AFAIK today "the Katello API for RHSM traffic" is really the RHSM API but implemented with a different authentication layer. The API is still the RHSM API.

Reply:
> AFAIK today "the Katello API for RHSM traffic" is really the RHSM API but implemented with a different authentication layer. The API is still the RHSM API.

My reasoning for being more specific is because this is not entirely true. This RHSM URL and the RHSM API we use is sort of combination of things. Every subscription-manager API call is supported by this API. Some APIs simply hit Katello's authorization and are forwarded to Candlepin. Some are intercepted, with intermediary processing before passing on to Candlepin. There may even be one or two that terminate at Katello. I get the idea of hiding the abstraction as much as possible. I'd balance that with the understanding that this `/rhsm` API is a wholly Katello made up concept (and technically not restricted solely to subscription-manager initiated connections).

Thus, I think "URL to the Katello API for RHSM traffic, or a reverse proxy to it" is pretty accurate. But, we can drop the `Katello' part in favor of 'URL to the API for RHSM traffic'. If ever we allow direct communication to Candlepin... I suspect a lot of other things will need to change. So we are bikeshedding a smidge :smile:

Reply:
> I'd balance that with the understanding that this `/rhsm` API is a wholly Katello made up concept (and technically not restricted solely to subscription-manager initiated connections).

I'm thinking from the standpoint that templates will generate subscription-manager commands but now that you mention it I don't recall whether there are any templates or other tooling that will send requests to /rhsm via cURL or python's requests library. I think that does happen with bootstrap.py which is not quite dead yet at least.

So I could reword to say that it isn't only requests originating from subscription-manager, though I do think that is what is most likely to make sense to users.